data/graph_db_utils.py

```diff
@@ -0,0 +1,44 @@
+import os
+from dotenv import load_dotenv
+from neo4j import GraphDatabase
+
+load_dotenv()
+
+URI = os.getenv("NEO4J_URI")
+AUTH = (os.getenv("NEO4J_USERNAME"), os.getenv("NEO4J_PASSWORD"))
+
+driver = GraphDatabase.driver(URI, auth=AUTH)
+
+def test_connection():
+    try:
+        driver.verify_connectivity()
+        print("database connection active")
+        
+    except Exception as e:
+        print(f"Error: {e}")
```

Comment:
It looks like when we get an exception, it will just print a message and then carry on. The calling code could think the request has succeeded. I suggest we'd want to throw an exception and stop execution in this case (using `raise` I think).

Same suggestion for other exception block